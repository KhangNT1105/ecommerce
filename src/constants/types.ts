export type MarketType = {
  countryCode: string;
  countryName: string;
  currencyCode: string;
  currencySymbol: string;
  active: boolean;
};

export type ClientType = {
  clientId: string;
  clientName: string;
  active: boolean;
};

export type PaginationType = {
  totalPages: number;
  totalItems: number;
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | undefined;
  size: number | 10;
  page: number | undefined;
};

export type MenuItem = {
  path: string;
  name: string;
  icon: string;
  children: MenuItem[];
};

export type ProductDenominationType = {
  label: string;
  price: number;
};

export interface IProductImage {
  logo: string;
  thumbnail?: string;
  portrait?: string;
  landscape?: string;
}

export interface ICategory {
  id: string;
  name: string;
  parentCategoryId?: string;
  externalCategoryId?: string;
  vendor?: string;
}

export interface IProductItem {
  id: string;
  displayName: string;
  vendor: string;
  productId?: string;
  market?: string;
  productBrand?: string;
  ghostLinkExpiryDays?: number;
  minimumAllowableExpiryDays?: number;
  valueType?: string;
  minPrice?: number;
  maxPrice?: number;
  redemptionType?: string;
  denominations?: string[];
  productDenominations?: ProductDenominationType[];
  ghostLinkInMarket?: number;
  allocated?: number;
  inQueue?: number;
  stockOnHand?: number;
  expiring?: number;
  internalName?: string;
  originalName?: string;
  description?: string;
  externalProductId?: string;
  categoryIds?: string[];
  published?: string;
  outOfStockBehaviour?: string;
  images?: IProductImage;
  brand?: {
    brandLogo?: string;
    brandName?: string;
  };
}

export interface AnyObject {
  [key: string]: any;
}

export interface StringPropertyObject {
  [key: string]: string;
}

export const SortValues = {
  ASC: 'updatedAt.asc',
  DESC: 'updatedAt.desc'
};

export const PortalType = {
  DOMAIN: 'domain',
  SMS: 'sms'
};

export type FnType = (param?: object) => void;
export type FnSubmitFormType = (event: React.FormEvent<HTMLFormElement>) => void;

export type EXPORT_TYPE = 'txt' | 'json' | 'csv' | 'xls' | 'xml';

export type PathParamsWithIDType = {
  id: string;
};
