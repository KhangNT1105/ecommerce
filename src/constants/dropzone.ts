import Configs from 'config';

export const UPLOAD_ACCEPT_FILE_TYPE: string = 'image/jpeg, image/jpg, image/png, image/svg, image/bmp, image/gif';
export const UPLOAD_ACCEPT_FILE_VIDEO_TYPE: string =
  'video/mp4, video/webm, video/wmv, video/flv, video/ogv, video/avi, video/mov';
export const FILE_DROPZONE_MIN_SIZE: number = 0;
export const FILE_DROPZONE_MAX_SIZE: number = 10048576;
export const FILE_DROPZONE_VIDEO_MAX_SIZE: number = 1000048576;
export const FILE_SIZE_IN_MB: number = 1;
export const FILE_SIZE_IN_PX: number = 1920;
export const RESOLUTION_MIN_SIZE_LANDSCAPE: number = 0;
export const RESOLUTION_MIN_SIZE_PORTRAIT: number = 0;

export const defaultDropzone = {
  apiURL: Configs.API.UPLOAD_SERVICE,
  files: [],
  autoUpload: true,
  config: {
    accept: UPLOAD_ACCEPT_FILE_TYPE,
    disabled: false,
    noClick: false,
    noKeyboard: false,
    minSize: FILE_DROPZONE_MIN_SIZE,
    maxSize: FILE_DROPZONE_MAX_SIZE
  },
  pushData: {
    error: false,
    msg: 'OK',
    infos: {
      title: 'Hero Banner Desktop and big screens',
      url: 'https://i.pinimg.com/236x/0a/8b/fb/0a8bfb162ef8a917f1e4d4bf5cd08e4c--android-apps-cover-art.jpg',
      size: '12222'
    }
  }
};

export const defaultDropzoneVideo = {
  ...defaultDropzone,
  config: {
    ...defaultDropzone.config,
    accept: UPLOAD_ACCEPT_FILE_VIDEO_TYPE,
    maxSize: FILE_DROPZONE_VIDEO_MAX_SIZE
  }
};

export const defaultDropzonePreviewInit = {
  files: [],
  onFileRemove: () => {},
  onFileUpload: () => {},
  status: false
};
