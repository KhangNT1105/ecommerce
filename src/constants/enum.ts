// TODO: All enum should be listed here
import { TypeOptions, ToastPosition } from 'react-toastify';
import { EXPORT_TYPE } from './types';

export const COLOR = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning',
  DANGER: 'danger',
  LINK: 'link'
};

export const SIZE = {
  LARGE: 'lg',
  SMALL: 'sm'
};

export const DURATION = {
  TOAST: 5000,
  TOAST_TRANSITION: 1000
};

interface POSITION_OPTIONS {
  [pos: string]: ToastPosition;
}
export const POSITION: POSITION_OPTIONS = {
  TOP_LEFT: 'top-left',
  TOP_RIGHT: 'top-right',
  TOP_CENTER: 'top-center',
  BOTTOM_LEFT: 'bottom-left',
  BOTTOM_RIGHT: 'bottom-right',
  BOTTOM_CENTER: 'bottom-center'
};

interface TYPE_OPTIONS {
  [type: string]: TypeOptions;
}
export const TOAST_TYPE: TYPE_OPTIONS = {
  SUCCESS: 'success',
  ERROR: 'error'
};

export const DATE_RANGE = {
  TODAY: 'today',
  LAST_7_DAYS: 'last7days',
  LAST_30_DAYS: 'last30days',
  LAST_WEEK: 'lastweek',
  LAST_MONTH: 'lastmonth',
  CUSTOM_DATE: 'customdate'
};

interface FILE_TYPE {
  [pos: string]: EXPORT_TYPE;
}
export const EXPORT_FILE_TYPE: FILE_TYPE = {
  CSV: 'csv',
  XLS: 'xls'
};

export const BREAKPOINTS = {
  EXTRA_LARGE: 1200,
  LARGE: 992,
  MEDIUM: 768,
  SMALL: 576
};

export const DROPZONE_EVENT_TYPE = {
  DROP: 'drop',
  CLICK: 'click'
};

export const ENTITIES = {
  PATH: {
    CONTENT: 'content',
    LICENSES: 'licenses',
    USER_MANAGEMENT: 'user-management',
    SYSTEM_CONFIGURATION: 'system-configuration'
  },
  NAME: {
    SHOWS: 'shows',
    SCHEDULES: 'schedules',
    CHANNELS: 'channels',
    CATALOGS: 'catalogs',
    CATEGORIES: 'categories',
    CASTINGS: 'casts',
    CLIPS: 'clips',
    PLAYLISTS: 'playlists',
    UPDATES: 'updates',
    CONTRACTS: 'contracts',
    LICENSORS: 'licensors',
    SYSTEM_USERS: 'system-users',
    USER_GROUPS: 'user-groups',
    SEASONS: 'seasons',
    CONFIGURATIONS: 'configuration',
    EPISODES: 'episodes',
    PAYMENT_GATEWAYS: 'payment-gateways',
    RISK_ENGINE_RULES_CONFIG: 'risk-engine'
  }
};

export const PUBLIC_PAGES = {
  NAME: {
    WELCOME: '/',
    LOGIN: 'login',
    TAC: 'term-and-conditions',
    ACCESS_DENIED: 'access-denied-403',
    DEMO: '/demo',
    PRODUCTS: 'sanpham',
    PRODUCT: 'sanpham/:id',
    CART: 'cart',
    CHECKOUT: 'checkout'
  }
};

export const LANGUAGES_CODE = {
  VN: 'vn',
  EN: 'en'
};

// This is API's entities from BE
export const API_ENTITIES_NAME = {
  SHOWS: 'shows',
  SCHEDULES: 'schedules',
  CHANNELS: 'channels',
  CATALOGS: 'catalogs',
  CATEGORIES: 'categories',
  CASTINGS: 'casts',
  CLIPS: 'clips',
  PLAYLISTS: 'playlists',
  UPDATES: 'updates',
  CONTRACTS: 'contracts',
  LICENSORS: 'licensors',
  SYSTEM_USERS: 'usergroups', // BE don't have entity's name : users | using same usergroups entity
  USER_GROUPS: 'usergroups',
  SEASONS: 'seasons',
  CONFIGURATIONS: 'configurations',
  EPISODES: 'episodes',
  RISK_ENGINES: 'riskengines'
};

export const PAGE_ACTIONS = {
  CREATE: 'create',
  EDIT: 'edit',
  DELETE: 'delete'
};

export const API_ERROR_CODE = {
  TOKEN_INVALID: '1012',
  TOKEN_EXPIRED: '1011',
  REFRESH_TOKEN_EXPIRED: '1014',
  REFRESH_TOKEN_INVALID: '1015'
};

export const PERMISSION_PREFIX = {
  VIEW: 'VIEW',
  CREATE: 'CREATE',
  EDIT: 'EDIT',
  EDIT_APPROVAL: 'EDIT_APPROVAL',
  EDIT_VIEW_CONTRACT: 'EDIT_VIEW_CONTRACT',
  ACTIVATE: 'ACTIVATE',
  DEACTIVATE: 'DEACTIVATE',
  APPROVE: 'APPROVE',
  REJECT: 'REJECT',
  DELETE: 'DELETE',
  RETRIEVE_BCM: 'RETRIEVE_BCM'
};

export const PAGE_ENTITY = {
  HOME: 'HOME'
};

export const BUTTONS_VISIBILITY = {
  SHOW: 'show'
};
export const PAGE_ACCESSIBILITY = {
  SHOW: 'show'
};

export const CLASS_NAME = {
  READONLY: 'read-only',
  DISABLED: 'disabled'
};

export const SEARCH_BOX_SUBMIT_DELAY = 600;

export const SHOWS_TYPES = {
  MOVIE: 'movie',
  SERIES: 'series',
  LIFE_EVENT: 'life event',
  THEATRICAL_PLAY: 'theatrical play',
  PROGRAM: 'program',
  MATCH: 'match',
  SEASONS: 'seasons',
  CHANNELS: 'channels',
  SCHEDULED: 'scheduled',
  SCHEDULING: 'ready for scheduling'
};

export const IMAGE_TYPES = {
  POSTER: 'poster',
  LANDSCAPE: 'landscape',
  POSTER_CLEAN: 'poster clean',
  LANDSCAPE_CLEAN: 'landscape clean',
  POSTER_WITH_LOGO: 'poster with logo',
  LANDSCAPE_WITH_LOGO: 'landscape with logo',
  LOGO_TITLE: 'logo title',
  HERO_BANNER_DESKTOP: 'hero banner desktop',
  HERO_BANNER_MOBILE: 'hero banner mobile'
};

export const INPUT_FIELD = {
  MAX_LENGTH_NORMAL: 100,
  MAX_LENGTH_URL: 1000
};

export const SORTING_DIRECTION = {
  ASCENDING: 'asc',
  DESCENDING: 'desc'
};
const SECONDS = {
  MINUTE: 60,
  HOUR: 3600,
  DAY: 86400,
  MONTH: 2592000,
  YEAR: 31536000
};
const ENUMS = {
  COLOR,
  SIZE,
  DURATION,
  POSITION,
  TOAST_TYPE,
  DATE_RANGE,
  EXPORT_FILE_TYPE,
  BREAKPOINTS,
  DROPZONE_EVENT_TYPE,
  SECONDS
};

export default ENUMS;
