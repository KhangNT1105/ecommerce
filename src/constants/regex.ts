const REGEX = {
  NUMBER_ONLY: /^[0-9]([.,][0-9])?$/,
  DIGIT_ONLY: /^[0-9]+$/
};

export default REGEX;
