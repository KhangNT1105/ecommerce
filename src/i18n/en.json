{
  "DASHBOARD_PAGE": "Welcome to Charity web",
  "LOGIN_PAGE": "Login",
  "SUBMIT": "Submit",
  "SIGN_IN": "Sign in",
  "FORM_ERROR_REQUIRED_FIELD": "This field is required",
  "FORM_ERROR_REQUIRED_FIELD_EMAIL_REQUIRED": "Email is required",
  "FORM_ERROR_REQUIRED_FIELD_EMAIL_INVALID": "Email is invalid",
  "FORM_ERROR_REQUIRED_FIELD_PASSWORD_REQUIRED": "Password is required",
  "FORM_ERROR_GTE_TO_FIELD": "This value must be greater than or equal to {{fieldLabel}}",
  "FORM_ERROR_LTE_TO_FIELD": "This value must be less than or equal to {{fieldLabel}}",
  "FORM_ERROR_ONLY_POSITIVE": "Only positive number is allowed",
  "FORM_ERROR_SERVER": "Username/password is incorrect",
  "FORM_ERROR_ALPHA_NUMBERIC_FIELD": "Only text and number are allowed",
  "FORM_ERROR_IS_URL_FIELD": "URL is invalid.",
  "FORM_ERROR_HOUR_FORMAT": "Hour format is invalid",
  "FORM_ERROR_MAX_VALUE": "This field cannot be greater than {{number}}.",
  "FORM_ERROR_MAX_LENGTH_NUMBER": "Maximum length of value should be {{number}} digits",
  "INVALID_DATA": "Value is invalid",
  "FORM_DATE_SHOULD_LESS_THAN_TO_DATE": "From Date should be less than To Date",
  "DATA_NOT_UNIQUE": "Value has already existed",
  "EXISTING_VALUE_ERROR": "{{value}} is already added. Please check again.",
  "EMAIL_LABEL": "Email Address",
  "PASSWORD_LABEL": "Password",
  "FORGOT_YOUR_PASSWORD": "Forgot your password?",
  "PLACEHOLDER_EMAIL": "e.g john@company.com",
  "PLACEHOLDER_PASSWORD": "Enter your password",
  "PLACEHOLDER_FILTER": "Start Searching to filter",
  "PLACEHOLDER_SEARCH_USER_GROUP": "Search user groups by group name or email",
  "PLACEHOLDER_SEARCH_USERS_IN_USER_GROUP": "Search users by email",
  "PLACEHOLDER_SEARCH_PLAYLIST": "Search playlist by ID, name, internalTitle, displayTitle",
  "PLACEHOLDER_TEXT": "Enter text here",
  "TERM_CONDITION": "Term & Condition",
  "SUPPORT": "Support",
  "COPYRIGHT": "Copyright Â© 2020 Acexis",
  "LOGOUT": "Logout",
  "LOGOUT_ERROR": "Logout failed",
  "MY_ACCOUNT": "My Account",
  "LOGO": "logo",
  "NO_DATA": "No result found.",
  "NO_RESULT": "No matches found.",
  "SAVE": "Save",
  "OK": "OK",
  "BACK": "Back",
  "CANCEL": "Cancel",
  "COPY": "Copy",
  "COPY_TO_CLIPBOARD": "Copy to clipboard",
  "COPIED": "Copied",
  "LOADING": "Loading...",
  "REGISTER_HERE": "Register",
  "ACEXIS_LOGO_ALT": "Acexis Logo",
  "VIDEO_PREVIEW": "Video Preview",
  "YES": "Yes",
  "NO": "No",
  "ATTENTION!": "Attention!",
  "ERROR!": "Error!",
  "SUCCESS!": "Success!",
  "WARNING": "Warning",
  "ADD_FILTER": "Add Filter",
  "EXPORT": "Export",
  "EXPORTING": "Exporting",
  "EXPORT_SUCCESS": "File is exported successfully",
  "EXPORT_FAILED": "Export failed",
  "SERVER_NOT_RESPONSE": "Server is not responded",
  "TODAY": "Today",
  "LAST_7_DAYS": "Last 7 days",
  "LAST_30_DAYS": "Last 30 days",
  "LAST_WEEK": "Last week",
  "LAST_MONTH": "Last month",
  "CUSTOM_RANGE": "Custom Range",
  "SELECT_RANGE": "Select Range",
  "ACTIVE": "Active",
  "IN_ACTIVE": "Inactive",
  "APPROVAL_PENDING_LABEL": "Approval Pending",
  "PENDING_LABEL": "Pending",
  "SVOD": "SVOD",
  "AVOD": "AVOD",
  "DRAFTS": "Drafts",
  "LABEL_REMEMBER": "Remember",
  "LOGIN_HEADER_TITLE": "Sign In to Application",
  "LOGIN_HEADER_TEXT": "Lorem ipsum dolor sit, amet consectetur adipisicing elit.",
  "LOGIN_FOOTER_TEXT": "Lorem ipsum dolor sit, amet consectetur adipisicing elit.",
  "ARTWORD_SUBTITLE": "Max size {{SIZE_IN_MB}} MB for each image and minimum resolution should be {{SIZE_IN_PX}}px<br />Only *.jpeg and *.png images will be accepted",
  "FILE": "FILE",
  "DRAG_AREA": "DRAG AREA",
  "POSTER": "POSTER",
  "LANDSCAPE": "LANDSCAPE",
  "UPLOAD": "Upload",
  "UPLOADING": "Uploading",
  "HERO_BANNER_MOBILE_LANDSCAPE": "Hero Banner Mobile - Landscape",
  "HERO_BANNER_DESKTOP_AND_BIG_SCREEN": "Hero Banner Desktop and big screens",
  "DROPZONE_MENTION_TEXT": "Upload Your files",
  "DROPZONE_IS_DRAG_ACTIVE": "Click here or drop a file to upload!",
  "DROPZONE_DROP_TO_FILE": "Drop it like it's hot!",
  "DROPZONE_IS_DRAG_REJECT": "File type not accepted, sorry!",
  "DROPZONE_IS_FILE_TOO_LARGE": "Maximum image size should be {{ maxSize }} Kb.",
  "DROPZONE_IS_FILE_RESOLUTION_TOO_SMALL": "Minimum resolution should be {{ resolutionMediaMinSize }}.",
  "DROPZONE_FILE_TYPE_NOT_ALLOWED": "File extension should be {{mimeType}}.",
  "DEFAULT_PHONE_NUMBER": "+ 00 123 456 789",
  "DEFAULT_GMAIL": "tuankhang@gmail.com",
  "LOGIN": "Login",
  "WISHLIST": "Wishlist",
  "SEARCH": "Search",
  "HOME": "Home",
  "INTRODUCTION": "Introduction",
  "PRODUCTS": "Products",
  "FREE_DELIVERY": "Free delivery",
  "WORLDWIDE": "Worldwide",
  "TWENTY_FOUR_SEVEN_SUPPORT": "24/7 support",
  "CUSTOMER_SUPPORT": "Customer support",
  "ALL_CATEGORIES": "All categories",
  "CONTACT": "Contact",
  "SECURE_SYSTEM": "Secure system",
  "ENUINE_PRODUCTS": "Enuine products",
  "CONTENT_REPAIR": "abc",
  "REPAIR_GUARANTEE": "Repair and Guarantee"
}
