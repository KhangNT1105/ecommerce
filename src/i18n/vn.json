{
  "TRANG_CHU": "Trang chủ",
  "GIOI_THIEU": "Giới thiệu",
  "SAN_PHAM": "Sản phẩm",
  "TIN_TUC": "Tin tức",
  "BAO_HANH": "Bảo hành",
  "HOI_DAP": "Hỏi đáp",
  "BAN_DO": "Bản đồ",
  "SAN_PHAM_MOI": "Sản phẩm mới",
  "MUA_NGAY": "Mua ngay",
  "UY_TIN_TRACH_NHIEM": "Uy tín trách nhiệm",
  "PHU_DE_UY_TIN_TRACH_NHIEM": "Uy tin trách nhiệm của công ty được đưa lên hàng đầu.",
  "VAN_CHUYEN_TOAN_QUOC": "Vận chuyển toàn quốc",
  "PHU_DE_VAN_CHUYEN_TOAN_QUOC": "Vận chuyển toàn quốc trong vòng 360 tiếng.",
  "MAY_BO_DAM_MOTOROLA": "Máy bộ đàm motorola",
  "DANG_NHAP": "Đăng nhập",
  "EMAIL_LABEL": "Email",
  "PASSWORD_LABEL": "Mật khẩu",
  "PLACEHOLDER_EMAIL": "Nhập email",
  "PLACEHOLDER_PASSWORD": "Nhập mật khẩu",
  "DANG_KY_TAI_DAY": "Đăng ký tại đây",
  "QUEN_MAT_KHAU": "Quên mật khẩu?",
  "FORM_ERROR_REQUIRED_FIELD": "Vui lòng nhập thông tin",
  "FORM_ERROR_REQUIRED_FIELD_EMAIL_REQUIRED": "Vui lòng điền email",
  "FORM_ERROR_REQUIRED_FIELD_EMAIL_INVALID": "Vui lòng điền email hợp lệ",
  "FORM_ERROR_REQUIRED_FIELD_PASSWORD_REQUIRED": "Vui lòng điền mật khẩu",
  "CHUA_CO_TAI_KHOAN": "Chưa có tài khoản? ",
  "DANG_KY": "Đăng ký",
  "TEN": "Tên",
  "HO_TEN_DEM": "Họ & tên đệm",
  "PLACEHOLDER_FIRST_NAME": "Nhập tên",
  "PLACEHOLDER_LAST_NAME": "Nhập họ và tên đệm",
  "FORM_ERROR_REQUIRED_FIELD_LAST_NAME_REQUIRED": "Vui lòng điền họ và tên đệm",
  "FORM_ERROR_REQUIRED_FIELD_FIRST_NAME_REQUIRED": "Vui lòng điền tên",
  "FORM_ERROR_REQUIRED_FIELD_PHONE_NUMBER_REQUIRED": "Vui lòng điền số điện thoại",
  "FORM_ERROR_REQUIRED_FIELD_CITY_REQUIRED": "Vui lòng chọn tỉnh/thành phố",
  "FORM_ERROR_REQUIRED_FIELD_DISTRICT_REQUIRED": "Vui lòng chọn quận/huyện",
  "FORM_ERROR_REQUIRED_FIELD_ADDRESS_REQUIRED": "Vui lòng điền địa chỉ",
  "FORM_ERROR_REQUIRED_FIELD_WARD_REQUIRED": "Vui lòng chọn phường/xã",
  "DANG_NHAP_TAI_DAY": "Đăng nhập tại đây",
  "DA_CO_TAI_KHOAN": "Đã có tài khoản? ",
  "FORM_ERROR_MIN_LENGTH_NUMBER": "Mật khẩu nên có độ dài ít nhất là {{value}} ký tự",
  "FORM_ERROR_CAPITAL_CHARACTER_FIELD_PASSWORD_REQUIRED": "Mật khẩu nên có ít nhất 1 ký tự in hoa",
  "SEARCH_PRODUCT_PLACEHOLDER": "Nhập tên sản phẩm...",
  "SAP_XEP_MAC_DINH": "Sắp xếp sản phẩm",
  "DANH_MUC_SAN_PHAM": "Danh mục sản phẩm",
  "MOI": "Mới",
  "DANG_XUAT_THANH_CONG": "Đăng xuất thành công",
  "SUCCESS!": "Thành công",
  "SUCCESS": "Thành công",
  "DANG_NHAP_THANH_CONG": "Đăng nhập thành công",
  "BAO_HANH_SUA_CHUA": "Bảo hành và Sửa chữa",
  "DANG_XUAT": "Đăng xuất",
  "LIEN_HE": "Liên hệ",
  "NHAN_HIEU": "Nhãn hiệu",
  "XUAT_XU": "Xuất xứ",
  "MODEL": "Model",
  "THEM_VAO_GIO_HANG": "Thêm vào giỏ hàng",
  "THONG_SO_KY_THUAT": "Thông số kỹ thuật",
  "BINH_LUAN": "Bình luận",
  "SAN_PHAM_TUONG_TU": "Sản phẩm tương tự",
  "DANH_GIA_GI_DO_VE_SAN_PHAM": "Đánh giá gì đó về sản phẩm...",
  "SAN_PHAM_PHO_BIEN": "Sản phẩm phổ biến",
  "GO_TO": "Đi tới",
  "SHOW_ROWS": "Hiện sản phẩm",
  "SHOW_RESULTS": "kết quả",
  "NO_DATA": "Không tìm thấy sản phẩm",
  "SORT_PRICE_HIGH_TO_LOW": "Giá từ cao đến thấp",
  "SORT_PRICE_LOW_TO_HIGH": "Giá từ thấp đến cao",
  "SORT_NEW_PRODUCT": "Sản phẩm mới",
  "SORT_OLD_PRODUCT": "Sản phẩm cũ",
  "TRA_LOI": "Trả lời",
  "HIEN_THI_THEM": "Hiển thị thêm",
  "XEM_THEM": "Xem thêm",
  "RUT_GON": "Rút gọn",
  "YEARS_AGO": "Năm trước",
  "MONTHS_AGO": "Tháng trước",
  "DAYS_AGO": "Ngày trước",
  "HOURS_AGO": "Giờ trước",
  "MINUTES_AGO": "Phút trước",
  "SECONDS_AGO": "Giây trước",
  "GIO_HANG": "Giỏ hàng",
  "GIA": "Giá",
  "SO_LUONG": "Số lượng",
  "TONG_TIEN": "Tổng tiền",
  "DIA_CHI_DETAILS": "256 Bùi Thị Xuân, Phường 3, Quận Tân Bình, TP. Hồ Chí Minh",
  "DIA_CHI": "Địa chỉ",
  "DIEN_THOAI": "Điện thoại",
  "SDT": "09 02 91 94 18",
  "EMAIL_DETAILS": "trungtamtracdia3@gmail.com",
  "THEM_VAO_GIO_HANG_THANH_CONG": "Thêm vào giỏ hàng thành công",
  "THANH_TOAN_GIO_HANG": "Thanh toán giỏ hàng",
  "THANH_TOAN": "Thanh toán",
  "UNAUTHENTICATION": "Bạn phải đăng nhập để có thể dùng chức năng này",
  "HAVE_CONTACT_PRODUCT_CONTENT": "Những sản phẩm liên hệ khi quý khách đặt hàng sẽ được nhân viên cửa hàng liên lạc để báo cáo về giá.",
  "CHI_TIET_HOA_DON": "Chi tiết hoá đơn",
  "PHONE_NUMBER": "Số điện thoại",
  "CITY": "Tỉnh/Thành phố",
  "DISTRICT": "Quận/huyện",
  "WARD": "Phường/xã",
  "ADDRESS": "Địa chỉ",
  "YOUR_ORDERS": "Đơn hàng của bạn",
  "NOTES": "Ghi chú thêm",
  "CHON_HINH_THUC_THANH_TOAN":"Chọn hình thức thanh toán",
  "THANH_TOAN_KHI_NHAN_HANG":"Thanh toán khi nhận hàng",
  "THANH_TOAN_BANG_TAI_KHOAN_NGAN_HANG":"Thanh toán bằng tài khoản ngân hàng"
}
