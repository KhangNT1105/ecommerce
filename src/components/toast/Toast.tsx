import React, { useEffect } from 'react';
import { Media } from 'reactstrap';
import { ToastContainer, TypeOptions, ToastPosition, toast, ToastContent, ToastOptions } from 'react-toastify';

import { TOAST_TYPE, POSITION } from '../../constants/enum';
import { classConcat } from '../../utils/common';
import i18n from 'i18n/i18n';

type Props = {
  type?: TypeOptions;
  message: string;
  position?: ToastPosition;
  duration?: number | false;
  handleClose?: () => void;
};

const Toast: React.FC<Props> = (props) => {
  // props
  const { type, duration, message, handleClose } = props;
  const position = props.position || POSITION.TOP_CENTER;

  const content: ToastContent = getContentToast(message, type);

  // states

  // on change type
  useEffect(() => {
    switch (type) {
      case TOAST_TYPE.ERROR:
        toast.error(content, { onClose: handleClose });
        break;
      case TOAST_TYPE.SUCCESS:
        toast.success(content, { onClose: handleClose });
        break;
      default:
        toast(content, { onClose: handleClose });
    }
  }, [type, content, handleClose]);

  return <ToastContainer position={position} autoClose={duration} draggable={false} hideProgressBar={true} />;
};

export const getIcon = (type?: string) => {
  let icon = 'fa-question';
  switch (type) {
    case TOAST_TYPE.ERROR:
      icon = 'fa-close';
      break;
    case TOAST_TYPE.SUCCESS:
      icon = 'fa-check';
      break;
  }
  return icon;
};
export const getTitle = (type?: string) => {
  let title = i18n.t('ATTENTION!');
  switch (type) {
    case TOAST_TYPE.ERROR:
      title = i18n.t('ERROR!');
      break;
    case TOAST_TYPE.SUCCESS:
      title = i18n.t('SUCCESS!');
      break;
    default:
      break;
  }
  return title;
};

export const getContentToast = (message: string, type?: TypeOptions) => ({
  closeToast
}: {
  closeToast: () => void;
}) => {
  return (
    <Media>
      <Media middle={true} left={true} className="mr-3">
        <i className={classConcat(['fa fa-fw fa-2x', `${getIcon(type)}`])} onClick={closeToast} />
      </Media>
      <Media body={true}>
        <Media heading={true} tag="h6">
          {getTitle(type)}
        </Media>
        <p>{message}</p>
      </Media>
    </Media>
  );
};

export const notify = {
  error: (message: string, options?: ToastOptions) => {
    return toast.error(getContentToast(message, TOAST_TYPE.ERROR), options);
  },
  success: (message: string, options?: ToastOptions) => {
    return toast.success(getContentToast(message, TOAST_TYPE.SUCCESS), options);
  }
};

export default Toast;
