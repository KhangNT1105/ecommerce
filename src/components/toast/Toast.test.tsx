import React from 'react';
import renderer from 'react-test-renderer';

import { DURATION, POSITION, TOAST_TYPE } from '../../constants/enum';
import Toast, { getTitle, getIcon, getContentToast } from './Toast';
import i18n from 'i18n/i18n';

describe('Toast is testing...', () => {
  test('Render toast successfully', () => {
    const toast = renderer.create(
      <Toast message={'This is default toast message.'} duration={DURATION.TOAST} position={POSITION.BOTTOM_RIGHT} />
    );

    expect(toast).toMatchSnapshot();
  });
});

describe('getTile', () => {
  test('it should return title is Attention', () => {
    expect(getTitle()).toEqual(i18n.t('ATTENTION!'));
  });
  test('it should return title is Error!', () => {
    expect(getTitle(TOAST_TYPE.ERROR)).toEqual(i18n.t('ERROR!'));
  });
  test('it should return title is Success!', () => {
    expect(getTitle(TOAST_TYPE.SUCCESS)).toEqual(i18n.t('SUCCESS!'));
  });
});

describe('getIcon', () => {
  test('it should return fa-question', () => {
    expect(getIcon()).toEqual('fa-question');
  });
  test('it should return fa-close', () => {
    expect(getIcon(TOAST_TYPE.ERROR)).toEqual('fa-close');
  });
  test('it should return fa-check', () => {
    expect(getIcon(TOAST_TYPE.SUCCESS)).toEqual('fa-check');
  });
});

describe('getContentToast', () => {
  const closeToast = jest.fn();
  test('test snapShot', () => {
    const content = renderer.create(getContentToast('success', 'success')({ closeToast }));

    expect(content).toMatchSnapshot();
  });
});
