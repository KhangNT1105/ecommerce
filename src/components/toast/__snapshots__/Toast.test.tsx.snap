// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast is testing... Render toast successfully 1`] = `
<div
  className="Toastify"
/>
`;

exports[`getContentToast test snapShot 1`] = `
<div
  className="media"
>
  <div
    className="mr-3 media-left media-middle"
  >
    <i
      className="fa fa-fw fa-2x fa-check"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="media-body"
  >
    <h6
      className="media-heading"
    >
      Thành công
    </h6>
    <p>
      success
    </p>
  </div>
</div>
`;
